source/d3d11/state_tracking.cpp

```diff
@@ -208,6 +219,60 @@ bool reshade::d3d11::state_tracking_context::update_depthstencil_clear_texture(D
 	return true;
 }
 
+std::vector<std::pair<ID3D11Texture2D*, reshade::d3d11::state_tracking::depthstencil_info>> reshade::d3d11::state_tracking::sorted_counters_per_used_depthstencil()
+{
+	struct pair_wrapper
+	{
+		int display_count;
+		ID3D11Texture2D* wrapped_dsv_texture;
+		depthstencil_info wrapped_depthstencil_info;
+		UINT64 depthstencil_width;
+		UINT64 depthstencil_height;
+	};
+
+	std::vector<pair_wrapper> sorted_counters_per_buffer;
+	sorted_counters_per_buffer.reserve(_counters_per_used_depth_texture.size());
+
+	for (const auto& [dsv_texture, snapshot] : _counters_per_used_depth_texture)
+	{
+		auto dsv_texture_instance = dsv_texture.get();
+		D3D11_TEXTURE2D_DESC desc;
+		dsv_texture->GetDesc(&desc);
+		
+		size_t texture_address = *((size_t*)(&dsv_texture_instance));
+		// get the display counter, if any of this texture.
+		if (_shown_count_per_depthstencil_address.count(texture_address) > 0)
```

Comment:
Would be better to use `.find() != .end()` here, since `.count()` doesn't cancel the lookup after the first find on `unordered_map`, at least on VC++ STL last I checked (I know that doesn't make much sense, considering that a map can only ever have one or zero entries for a key, but it is what it is).

Reply:
Changed. :)